Migrate PHPUnit Tests and Use WordPress test lib instead of WP Mocks

ravinderk opened the pull request:
<!--
Filling out this template is required.  Any PR that does not include enough information to be reviewed may be closed at a maintainers' discretion.  All new code requires documentation and tests to ensure against regressions.
-->
🔴 This pull request is not ready for review.

### Description of the Change
<!--
We must be able to understand the design of your change from this description.  The maintainer reviewing this PR may not have worked with this code recently, so please provide as much detail as possible.

Where possible, please also include:
- verification steps to ensure your change has the desired effects and has not introduced any regressions
- any benefits that will be realized
- any alternative implementations or possible drawbacks that you considered
- screenshots or screencasts
-->

In a few cases, We found that Mocking WordPress (`10UP\WP_Mock`) for tests does not help to produce a reliable build. We decided to implement PHPUnit tests with WordPress.

<!-- Enter any applicable Issue number(s) here that will be closed/resolved by this PR. -->
Closes #

### How to test the Change
<!-- Please provide steps on how to test or validate that the change in this PR works as described. -->
- Open the terminal and switch to the plugin root directory
- Run `composer install`
- Install WordPress and create a database `tests/bin/install-wp-tests.sh distributor_tests root root`
- Run PHPUnit test ` vendor/bin/phpunit`

### Changelog Entry
<!--
Please include a summary for this PR, noting whether this is something being Added / Changed / Deprecated / Removed / Fixed / or Security related.  You can replace the sample entries after this comment block with the single changelog entry line for this PR. -->
> Added - Migrate PHPUnit tests to WordPress from`10UP/WP_Mock`


### Credits
<!-- Please list any and all contributors on this PR so that they can be added to this projects CREDITS.md file. -->
Props @ravinderk 


### Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you are unsure about any of these, please ask for clarification.  We are here to help! -->
- [ ] I agree to follow this project's [**Code of Conduct**](https://github.com/10up/.github/blob/trunk/CODE_OF_CONDUCT.md).
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my change.
- [ ] All new and existing tests pass.

ravinderk committed:
dev: add initial commit to migrate from mock phpunit test

ravinderk committed:
doc: set correct return type

ravinderk committed:
refactor: mark mocked tests incomplete

ravinderk committed:
dev: update test github workflow

jeffpaul commented:
@ravinderk any chance of getting this wrapped up in the next week or so to be ready for the v2 release?

ravinderk committed:
Merge branch 'develop' into feat/migrate-phpunit

ravinderk committed:
format: improve code formatting

ravinderk committed:
refactor: change file location

ravinderk committed:
format: remove unnecessary php doc tags from class comment

ravinderk committed:
tests: update DistributorPostTest

ravinderk committed:
format: improve code formatting

ravinderk committed:
Merge branch 'trunk' into feat/migrate-phpunit

ravinderk committed:
chore: remove unnecessary file

Commit:
add: create README.md file for tests directory

Commit:
add: update file name

peterwilsoncc committed:
Merge branch 'develop' into feat/migrate-phpunit

# Conflicts:
#	tests/php/DistributorPostTest.php
#	tests/php/NetworkSiteConnectionsTest.php
#	tests/php/WordPressExternalConnectionTest.php
#	tests/php/bootstrap.php

jeffpaul commented:
@10up/open-source-practice this would be a good PR to help pick up and get finished off

faisal-alvi committed:
add tests for `NetworkSiteConnection` class

faisal-alvi committed:
Merge branch 'develop' into feat/migrate-phpunit

faisal-alvi committed:
remove already defined consts,
https://github.com/10up/distributor/actions/runs/10097420348/job/27922230463?pr=1073

faisal-alvi commented:
Update:

- resolved conflicts
- fixed PHPUnit test failure issues
- fixed already defined constants issue
- added tests for `NetworkSiteConnection` class
- ignore E2E failures, they're being addressed in another PR

Note: I'm not assigning this to myself to allow others to pick it up and work on the remaining files

peterwilsoncc commented on includes/classes/DistributorPost.php:
```diff
@@ -27,25 +27,6 @@
  * shown in IDEs.
  *
  * @since 2.0.0
- *
- * @method bool has_blocks()
- * @method bool has_block( string $block_name )
- * @method int  get_the_ID()
- * @method string get_permalink()
- * @method string get_post_type()
- * @method int|false get_post_thumbnail_id()
- * @method string|false get_post_thumbnail_url( string $size = 'post-thumbnail' )
- * @method string|false get_the_post_thumbnail( string $size = 'post-thumbnail', array $attr = '' )
- * @method string get_canonical_url( string $canonical_url = '' )
- * @method string get_author_name( string $author_name = '' )
- * @method string get_author_link( string $author_link = '' )
- * @method array get_meta()
- * @method array get_terms()
- * @method array get_media()
- * @method array post_data()
- * @method array to_insert( array $args = [] )
- * @method array to_pull_list( array $args = [] )
- * @method array to_rest( array $args = [] )
```
This block is required to allow IDEs and docblock parsers to pick up the methods. They are all marked private so we can use a magic method to call them after ensuring the correct site is current on multisite.
